prov/tcp/src/xnet_eq.c

```diff
@@ -158,57 +128,62 @@ static int xnet_eq_wait_try_func(void *arg)
 	return FI_SUCCESS;
 }
 
-int xnet_eq_add_progress(struct xnet_eq *eq, struct xnet_progress *progress,
-			 void *context)
+/* Safe to call even if domain was never added */
+static void xnet_eq_del_domain(struct xnet_eq *eq, struct xnet_domain *domain)
 {
-	return ofi_wait_add_fd(eq->util_eq.wait,
-			       ofi_dynpoll_get_fd(&progress->epoll_fd),
-			       POLLIN, xnet_eq_wait_try_func, NULL, context);
-}
+	assert(!xnet_progress_locked(&domain->progress));
```

Comment:
Can you remove this one as well?

Reply:
Updated -- I checked and didn't see anywhere else in the provider where I have a similar check for !locked.

Reply:
Thank you!